app/collins/softlayer/SoftLayer.scala

```diff
@@ -94,10 +92,13 @@ object SoftLayer extends SoftLayer {
 
   override def activateServer(id: Long): Future[Boolean] = {
     val url = softLayerUrl("/SoftLayer_Hardware_Server/%d/sparePool.json".format(id))
-    val query = JsObject(Seq("parameters" -> JsArray(List(JsString("activate")))))
-    val queryString = Json.stringify(query)
+    val query = Json.obj(
+      "parameters" -> Json.arr(
+         "activate"
+      )
+    )
 
-    val wsUrl = WS.url(url.toString).withHeaders("Content-Type" -> "application/json", "Content-Length" -> queryString.length.toString)
```

Comment:
Play will make an in-memory buffer and pass it to ning (https://github.com/playframework/playframework/blob/master/framework/src/play-ws/src/main/scala/play/api/libs/ws/WS.scala#L398)

Ning will get the Content-Length part (because http/1.1 says it needs a content-length if not streamed (which is not the case here)). 

The Content-Type is set by play because you pass in a JsValue type and Content-Type will automatically resolve to 'application/json' **compile-time**. 
Implicit / call-chain is:
 * https://github.com/playframework/playframework/blob/master/framework/src/play-ws/src/main/scala/play/api/libs/ws/WS.scala#L402
 * https://github.com/playframework/playframework/blob/master/framework/src/play/src/main/scala/play/api/http/Writeable.scala#L94
 * https://github.com/playframework/playframework/blob/master/framework/src/play/src/main/scala/play/api/http/ContentTypeOf.scala#L51
 * https://github.com/playframework/playframework/blob/master/framework/src/play/src/main/scala/play/api/http/StandardValues.scala#L29
 * https://github.com/playframework/playframework/blob/master/framework/src/play/src/main/scala/play/api/http/StandardValues.scala#L110

Related documentation: https://www.playframework.com/documentation/2.3.x/ScalaWS#Submitting-JSON-data